### PR TITLE
App Services CLI

### PR DESCRIPTION
**Pull request by nlarew**
## Pull Request Info

- Moves the generated Realm CLI docs into a DEPRECATED state under a new URL path (`/cli` -> `/realm-cli`)
- Adds new generated App Services CLI docs at `/cli`
- Introduces new source constants to refer and link to `app-services-cli` reference pages throughout the docs

### Jira

- (DOCSP-25469) Generate command reference docs for app-services-cli
- (DOCSP-25470) [BAAS] Rename realm-cli to app-services-cli

### Staged Changes (Requires MongoDB Corp SSO)

- [Staged](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/cli/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: snooty.toml [hunk 1]
```diff
@@ -63,8 +63,10 @@ backend-short = "App Services"
 base-url = "https://www.mongodb.com/docs/atlas/app-services"
 charts = "MongoDB Charts"
 charts-short = "Charts"
-cli = "Realm CLI"
-cli-bin = ":binary:`realm-cli`" # binary -- DO NOT USE IN LINKS! Will break them.
+cli = "App Services CLI"
+cli-bin = "app-services-cli"
+cli-bin-mono = "``app-services-cli``"
+cli-bin-ref = ":ref:`app-services-cli <app-services-cli>`"
```
Comment (mongodben) [inline snooty.toml:69]: why make these 2?

**Review by mongodben — COMMENTED** [review]
why replace text with source constants? do you anticipate a future additional name change?

File: source/schemas/enforce-a-schema.txt [hunk 2]
```diff
@@ -249,25 +249,25 @@ Procedure
...
             .. code-block:: bash
 
-               realm-cli login --api-key="<MongoDB Cloud Public API Key>" --private-api-key="<MongoDB Cloud Private API Key>"
+               {+cli-bin+} login --api-key="<MongoDB Cloud Public API Key>" --private-api-key="<MongoDB Cloud Private API Key>"
 
 
          .. step:: Pull the Latest Version of Your App
 
-            To define a document schema with {+cli-bin+}, you need a local copy of your
+            To define a document schema with {+cli-bin-mono+}, you need a local copy of your
```
Comment (mongodben) [inline source/schemas/enforce-a-schema.txt:263]:
```suggestion
            To define a document schema with the {+cli+}, you need a local copy of your
```

nit

File: source/template-apps.txt [hunk 3]
```diff
@@ -108,7 +108,7 @@ Create a Template App
 
          If you want to use the iOS with UIKit template app, select :guilabel:`SwiftUI + Atlas Device Sync Starter`
          in the list of template apps, but specify ``ios.swift.todo`` when you pull the
-         client code using the {+cli-bin+}.
+         client code using {+cli-bin-mono+}.
```
Comment (mongodben) [inline source/template-apps.txt:111]:
```suggestion
         client code using the {+cli+}.
```

nit

File: snooty.toml [hunk 4]
```diff
@@ -63,8 +63,10 @@ backend-short = "App Services"
 base-url = "https://www.mongodb.com/docs/atlas/app-services"
 charts = "MongoDB Charts"
 charts-short = "Charts"
-cli = "Realm CLI"
-cli-bin = ":binary:`realm-cli`" # binary -- DO NOT USE IN LINKS! Will break them.
+cli = "App Services CLI"
+cli-bin = "app-services-cli"
+cli-bin-mono = "``app-services-cli``"
```
Comment (mongodben) [inline snooty.toml:68]: stylistically i think this should pretty much never be used. i understand that you're substituting existing similar text here (i.e `realm-cli`), but i think most all of these instances should be replaced by just the "App Services CLI".

**Comment by mongodben** [comment]
i reviewed, though didn't read thru literally every change. couple of higher level questions about the use of source constants above, and some small nitty changes

**Comment by nlarew** [comment]
Per our team discussion, I moved this away from `cli-bin-mono` and `cli-bin-ref` to use the non-monospace name.

> why replace text with source constants? do you anticipate a future additional name change?

I do - eventually I can see this merging into Atlas CLI. The main goal here is to make all the spots we use it easier to find with just one or two greps.

File: source/includes/app-services-cli-overview.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
+.. _cli-overview:
```
Comment (mongodben) [inline source/includes/app-services-cli-overview.rst:1]: why is this an include now? based on cursory look, seems like it's only used 1 place

**Review by mongodben — APPROVED** [review]
LGTM. didn't review literally everything, but generally seemed good. and don't know how a significant error could pop up here. 

1 question abt why you made something into an include file as well

File: source/includes/app-services-cli-overview.rst (hunk 5 above)
Comment (nlarew) [reply source/includes/app-services-cli-overview.rst:1]: This will make it easier to regenerate the page

**Comment by nlarew** [comment]
Picking up in https://github.com/mongodb/docs-app-services/pull/594

**Pull request closed without merging**